### DISCUSSION TITLE
Commas in nested register name

### DISCUSSION THREAD

**Discussion by colecon** [react-hook-form/react-hook-form#6596] [Q&A]
After migrating from v6 to v7, one of my forms had broken. I kept receiving an error on render `TypeError: Cannot read properties of undefined (reading '_f')`

After much debugging i figured out that it came from the register and the specific name i was giving it:
{...register(`goals.${newStringResponse}`) where `newStringResponse = 'option41/22,option32/63'`

I want the formData structure to look like so:
```
formData = {
  user: 'ujsy-ds4faA',
  goals: {
    'option41/22,option32/63': 25,
    'option42/27,option24/33': 20
    ...
  }
}
```
This worked before the v7 update but now it seems that i am not allowed to use a comma in the key name anymore. Is this true? is there a workaround?

These are the last 2 stack frames

```
registerFieldRef | @ | createFormControl.ts:945
ref | @ | createFormControl.ts:1016
```

**Comment by slugmandrew** [comment]
Can't comment on the internals of v6 vs v7, but a simple solution would be to urlencode the data with the commas in it, and then decode it back afterwards. At least you'll guarantee safety then.

Possibly telling you things you already know but `option41/22,option32/63` would be converted to `option41%2F22%2Coption32%2F63`.

Perhaps this is a bug / oversight in the library in which case it might need to be fixed internally, where I imagine a similar fix would suffice.

Perhaps @bluebill1049 can comment?
